### ISSUE TITLE
Ancient Apparation bug

### ISSUE THREAD

**Issue by WasBooking** [jeffhill/Dota2#265]
### Ability name

Ice Blast

### Description

video: https://www.youtube.com/watch?v=VW62lFdL_f0

### Example Match ID (and possibly Timestamp)

video

**Comment by NMOTH** [comment]
What is the bug here? BKB going on CD in demo mode? is it only doing this on storm? has this happened in an actual match ( if yes provide ID). Note some behaviors are different demo mode ( which is a separate issue on its own)

**Comment by TheUnholyWrath** [comment]
So Storm gets hit by Ice Blast, Storm gets stunned by Ravage, Storm queues up BKB, Storm uses BKB, Ice Blast kills Storm. 

Ice Blast kills through BKB so unless I'm missing something theres no bug here

**Comment by NMOTH** [comment]
Yes Ice blast kills through bkb. This is not a bug. Please refer to ability description as well. 
![image](https://user-images.githubusercontent.com/106769588/171759661-b5b2fb85-0fd9-4e52-a68a-2f9419309adf.png)

please close this thanks

**Comment by WasBooking** [comment]
> Yes Ice blast kills through bkb. This is not a bug. Please refer to ability description as well. ![image](https://user-images.githubusercontent.com/106769588/171759661-b5b2fb85-0fd9-4e52-a68a-2f9419309adf.png)
> 
> please close this thanks

storm dies in stun, he can't press bkb in any way, but it goes into cd, i'll close it

**Comment by TheUnholyWrath** [comment]
Yeah Ice Blast shatter is letting you BKB when you're dead. I was wrong before, was testing it out and didn't look at it again before posting. Think of it like if you just die to ravage there, you try to queue up BKB but you die before you can use it, shatter isn't resolving correctly. You're dying to ravage damage but still getting the BKB off afterwards

**Comment by WasBooking** [comment]
> Yeah Ice Blast shatter is letting you BKB when you're "dead". I was wrong before, was testing it out and didn't look at it again before posting. Think of it like if you just die to ravage there, you try to queue up BKB but you die before you can use it, shatter isn't resolving correctly. You're shattering to ravage damage but still getting the BKB off before you actually die

so is this a bug or not?

**Comment by TheUnholyWrath** [comment]
![20220603102138_1](https://user-images.githubusercontent.com/46859977/171761182-d37f34bd-cc07-4b75-86e7-e7765f76e759.jpg)

**Comment by TheUnholyWrath** [comment]
Ravage shouldn't be getting removed before Ice Blast damage. Seems like we lose the ravage debuff because we are in range to shatter, but because we're not stunned anymore we can BKB.

**Comment by isRobertMa** [comment]
This is an interesting one, it is certainly worth looking into. After all, the logical sense is you take ravage and die, since you are stunned during the whole duration you shouldn't be able to use BKB. I think this might be an example of the BKB "insert game tick" mechanic going the wrong example. There is a Reddit post about the inconsistency you can check right [here](https://www.reddit.com/r/learndota2/comments/s0j0dv/black_king_bar_mechanics_information_things_to_be/), mechanically wise it does exactly what it suppose to do, however, in this situation, the BKB tick insert itself between **ending the stun** and **the tick you die**, thus we have this frustrating experience.
